packages/shared/src/components/auth/RegistrationForm.tsx

```diff
@@ -58,7 +62,33 @@ export const RegistrationForm = ({
 }: RegistrationFormProps): ReactElement => {
   const { trackEvent } = useContext(AnalyticsContext);
   const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const [username, setUsername] = useState<string>('');
+  const [name, setName] = useState<string>('');
   const isAuthorOnboarding = trigger === AuthTriggers.Author;
+  const { requestMethod } = useRequestProtocol();
+  const usernameQueryKey = ['generateUsername', name];
+  useQuery<{
+    generateUniqueUsername: string;
+  }>(
+    usernameQueryKey,
+    () =>
+      requestMethod(
+        graphqlUrl,
+        GET_USERNAME_SUGGESTION,
+        { name },
+        { requestKey: JSON.stringify(usernameQueryKey) },
+      ),
+    {
+      enabled: !!name.length,
+      onSuccess: (data) => {
+        if (!data.generateUniqueUsername.length || !!username.length) return;
+
+        if (data?.generateUniqueUsername) {
+          setUsername(data.generateUniqueUsername);
+        }
+      },
+    },
```

Comment:
It is actually an anti pattern to use this callback since it is being removed from `useQuery` in future versions of `react-query`. https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose

They actually propose `useEffect` as alternative, I realize it was something @rebelchris commented against but not sure what other thing we could use. Hopefully now that we merged https://github.com/dailydotdev/apps/pull/1745 we can at least use it more reliably.

Reply:
Ah sorry my bad, wasn't aware of this, but thanks for raising this.

So my suggestion when we switch to a hook we can leverage useEffect again.
@capJavert Maybe you can help tom with this solution if you agree with this?

Reply:
Talked with @rebelchris, `useEffect` should be fine here to sync up the generate username but for sure let's move whole logic to hook to keep it contained.

Reply:
Oh, wow. That is a huge change. Thanks for letting us know about it.